extensions/1password/src/v8/utils.ts

```diff
@@ -58,30 +58,56 @@ export function op(args: string[]) {
   throw Error("1Password CLI is not found!");
 }
 
+const CACHE_TIMEOUTS: { [key: string]: number } = {
+  [CATEGORIES_CACHE_NAME]: 1000 * 60 * 10,
+  [ITEMS_CACHE_NAME]: 1000 * 60 * 10,
+  [ACCOUNT_CACHE_NAME]: 1000 * 60 * 10,
+};
+
+const DEFAULT_CACHE_TIMEOUT = 1000 * 60 * 10;
+
 export function useOp<T>(args: string[], cacheKey?: string) {
   const [data, setData] = useState<T>();
   const [error, setError] = useState<unknown>();
   const [isLoading, setIsLoading] = useState<boolean>(true);
 
   useEffect(() => {
-    if (cacheKey && cache.has(cacheKey)) {
-      setIsLoading(false);
-      return setData(JSON.parse(cache.get(cacheKey) as string));
-    }
-
-    try {
-      const items = op([...args, "--format=json"]);
+    const fetchData = async () => {
+      setIsLoading(true);
+      try {
+        const items = op([...args, "--format=json"]);
+        if (cacheKey) {
+          cache.set(cacheKey, items);
+          cache.set(`${cacheKey}_lastUpdated`, Date.now().toString());
+        }
+        console.log(`fetch ${cacheKey} data success`);
+        setData(JSON.parse(items));
+      } catch (error: unknown) {
+        setError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    };
 
-      if (cacheKey) {
-        cache.set(cacheKey, items);
-        return setData(JSON.parse(cache.get(cacheKey) as string));
+    if (cacheKey && cache.has(cacheKey)) {
+      const lastUpdatedString = cache.get(`${cacheKey}_lastUpdated`) as string;
+      const lastUpdated = parseInt(lastUpdatedString, 10);
+      const timeSinceLastUpdate = Date.now() - lastUpdated;
+      const cacheTimeout = CACHE_TIMEOUTS[cacheKey] || DEFAULT_CACHE_TIMEOUT;
+
+      console.log(
+        `${cacheKey} last udpated is: ${lastUpdated}, time since last update: ${timeSinceLastUpdate}, specify cache timeout: ${cacheTimeout}`
```

Comment:
typo

```suggestion
        `${cacheKey} last updated is: ${lastUpdated}, time since last update: ${timeSinceLastUpdate}, specify cache timeout: ${cacheTimeout}`
```

Reply:
These logs will be removed